### ISSUE TITLE
[Issue]: Audio switching for files that work perfectly for 10.0.7.0 doesn't work 

### ISSUE THREAD

**Issue by luky92** [jellyfin/jellyfin#7896]
### Please describe your bug

Changing audio stream in a new version of jellyfin makes it stick to the first choosen audio stream 

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS:Linux
- Virtualization:Docker (linuxserver image)
- Clients:Browser
- Browser:Chrome 
- FFmpeg Version:included in docker 
- Playback Method: Unknown 
- Hardware Acceleration:none
- Plugins:none 
- Reverse Proxy:yes 
- Base URL:-
- Networking:-
- Storage:-
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by crobibero** [comment]
Which client are you experiencing this with? Only chrome browser?

**Comment by luky92** [comment]
Whatever the standard web client is i and a friend had the same problem
can't test it anymore went back to 10.7 after expiriencing the problem i
forgot to mention that i use reverse proxy
W dniu niedz., 12.06.2022 o 18:18 Cody Robibero ***@***.***>
napisał(a):

**Comment by crobibero** [comment]
I can't reproduce this using Firefox on 10.8

**Comment by luky92** [comment]
The problem was present on chrome didn't check Firefox
W dniu niedz., 12.06.2022 o 18:34 Cody Robibero ***@***.***>
napisał(a):

**Comment by cvium** [comment]
Make sure you clear the cache

**Comment by luky92** [comment]
![image](https://user-images.githubusercontent.com/15126111/173244561-fcf792c7-e5b3-4251-a388-173016dbaeb0.png)
here is a screenshot of the problematic video i wonder what does the audio media handler mean

**Comment by silentTeee** [comment]
I experience this as well, even after clearing cache, on a native install with no virtualization

**Comment by luky92** [comment]
any updates on this?

**Comment by gomaaz** [comment]
Same issue here. Standard audio stream is playing, switching is impossible -> playback error

**Comment by jcdick1** [comment]
On my Roku, any media with multiple audio tracks added after the 10.8 update had issues playing the alternate audio tracks.  For media with multiple audio tracks added pre-10.8.0, in the JF logs, the entries for "StreamBuilder.BuildVideoItem" would use either "AudioStreamIndex=Null" or "AudioStreamIndex=2" for the second audio track selected and that second audio track was played.

Files added after the update were using "AudioStreamIndex=3" even though there are only two audio tracks in the file, and the second track doesn't play, instead it plays the first track or default track.

**Comment by luky92** [comment]
any updates on this this issiue is still present in the newestr release if any extra info is neede please ask I can provide anything

**Comment by luky92** [comment]
any updates on this can someone please take a look? :)

**Comment by luky92** [comment]
figured out what the problem is jellyfin has a problem handling aac codec this results in default audio being the only returned stream

**Comment by luky92** [comment]
workaround is to convert audio to eac3 codec with a comment similar to this ffmpeg -i input.mkv -map 0 -c:v copy -c:a eac3 -c:s copy output.mkv

**Comment by rangeroob** [comment]
That is my findings are well with mp4 files, I have found that you can have the first audio track be acc but all other ones need to be ac3 or another format that transcodes/Direct Streams the audio stream to acc. I am only having this issue on the Roku and the web player for Android.

**Comment by luky92** [comment]
Still happens

sob., 19 sie 2023, 03:47 użytkownik jellyfin-bot ***@***.***>
napisał:

**Comment by d86leader** [comment]
Still happens, randomly. I got this in web client in firefox, and in swiftfin, but couldn't reproduce the next day (server was up the whole time)

**Comment by ParasiteDelta** [comment]
Yep, still a thing. Tested using a completely fresh Jellyfin install (fresh as in, down to the OS), the Jellyfin app on Android 12, and even using the app available for Roku. I'm just more puzzled than frustrated on how this is such a big issue here and not in Emby, which has had this solved for at least two to three years. From what I could vaguely find, seems to be something to do with the server and its method of indexing available audio/subtitle/resource tracks?

I do love this fork, as it actually plays all of my media at full bitrate without needless transcoding and without chugging or hitching unlike Emby, but the inability to actually select your audio tracks is a massive deal-breaker, since this affects everything from anime to foreign films to some cartoons. And yes, asking people to set up remuxing pipelines for multiple series is ridiculous.

**Comment by ParasiteDelta** [comment]
Found a Reddit post mentioning this issue back in 2021, and sure enough, they're right; the main issue comes with how the server handles audio tracks in the AAC format. Whether you remux the video file with AC3 conversions or just extract the AAC tracks and convert them to AC3, the issue disappears if you convert the audio tracks to AC3.

Considering how tedious it would be to solve this by hand, and the fact that you have to solve this by hand as an end-user, I might honestly just suffer and go back to Emby. This is ridiculous.

**Comment by iron-mors** [comment]
That's sad... It seemed that Jellyfin will be a good replacement for Kodi for me. But not being able to switch audio is just a dealbreaker.
It's nice that a workaround exists to reencode - but this is far from an ideal solution of "just works".
Hopefully this will be resolved and I can recommend jellyfin again as a go-to solution.
Fingers crossed.

**Comment by steamboy23** [comment]
Same here

**Comment by gnattu** [comment]
Does anybody here has any samples that confirmed not working and is legal to share with? I tried to reproduce this but my files are switching just fine, by looking at reddit it seems like it only happens for specific video files.

**Comment by luky92** [comment]
Seems like double AAC audio would do the trick

**Comment by gnattu** [comment]
I tried multiple files with two, three, and four AAC audio tracks, and none of them have the switching issue. So it only happens with specific files, but I don't have any in my hand at the moment.

**Comment by maurges** [comment]
Sorry, I couldn't even reproduce it with my ~~illegal~~ files that it triggered on randomly. Playing the same file again I can switch audio no problem.

One detail: it's not necessarily AAC, here's an excerpt from ffprobe:
```
  Stream #0:1(eng): Audio: dts (DTS), 48000 Hz, 5.1(side), fltp, 768 kb/s (default)
  Stream #0:2(rus): Audio: dts (DTS), 48000 Hz, 5.1(side), fltp, 768 kb/s
```

**Comment by spagootie** [comment]
still happens

**Comment by Shadowghost** [comment]
@spagootie please provide a short sample of the failing file and the exact codec config

**Comment by luky92** [comment]
issiue is back 
here is the scrennshoot of ofending file metadata 
![Image](https://github.com/user-attachments/assets/fb1524b8-358b-4641-b7e1-57bf9570a751)

**Comment by luky92** [comment]
looks like the issiue is 5.1 audio cfg @gnattu

**Comment by luky92** [comment]
problem still exists however i noticed it only happens on chrome base d browswers works perfectly on firefox

**Comment by luky92** [comment]
so it could be web issiue

**Comment by felix920506** [comment]
If you have a file that causes the issue, please start a separate fresh instance and see if it happens. If it still happens, please provide the sample file.

**Comment by vilim** [comment]
I have this issue as well (only on the samsung/Tizen though, all other devices work well)

**Comment by luky92** [comment]
I eventually managed to generate a test video that reproduces this issue 

https://github.com/user-attachments/assets/b0efea9f-f935-4e38-8b95-75670a587524
here is the file ( carefull with volume one audio track is static another tones ) issiue also reproduces on clean windows intall

**Comment by luky92** [comment]
Would be nice if you didn't have to do that of course:)

czw., 24 kwi 2025, 22:48 użytkownik gnattu ***@***.***>
napisał:

**Comment by gnattu** [comment]
Well, the things is in post 10.8 clients we are trying to do the audio track selection exclusively on the browser side if possible, but that feature is behind the chrome feature flag, and for some reason, the fallback behavior to use the server side remux does not work which makes you to get locked in the default stream.

On firefox this file is played with direct stream and uses server side remux and that is why stream selection still works. When you watch this file on Chrome, there will be no ffmpeg logs generated, but there will be multiple when you are playing this file on firefox.

**Comment by gnattu** [comment]
I will close this for now in favor of the web side issue.
